HuliWebShop/src/main/java/huli/example/huliwebshop/controllers/ProductController.java

```diff
@@ -21,14 +26,42 @@ public ResponseEntity getAllProduct(){
             return ResponseEntity.status(400).body(e.getMessage());
         }
     }
+
     @GetMapping("/all-product/{id}")
-    public ResponseEntity editBuilding(@PathVariable Long id){
+    public ResponseEntity getProduct(@PathVariable Long id){
         try {
             return ResponseEntity.ok().body(productService.getByAlone(id));
         } catch (Exception e) {
             return ResponseEntity.status(400).body(e.getMessage());
         }
     }
 
+    @PostMapping("/admin/create")
```

Comment:
REST routing should be fixed here too: "admin" is a role (i'd assume), not a path, so shouldn't it be validated if the user is really an admin? Create endpoints should be simply a POST /api/{resource-to-be-created}, in this case POST /api/products. Editing is PUT /api/products/{id} and deleting is DELETE /api/products/{id}

Reply:
...or another common pattern is if the admin is considered a different api that is available for multiple roles (but since it's contains "administrative" functions, it is separated). In that case the routes could be organized in a way like this:

GET /api/public/products -> products list
GET /api/public/products/{id} -> single
GET /api/public/orders -> orders list
...
POST /api/admin/products -> create
PUT /api/admin/products/{id} -> update
DELETE /api/admin/products/{id} -> delete

In this case it's also possible to validate users by patterns, for example check if on the /api/admin/* the users should have X or Y roles.